### PR TITLE
Reviews: fix reviews without rating not appearing when sorting by rating

### PR DESCRIPTION
**Pull request by Aljullu**
### How to test the changes in this Pull Request:

1. Create a review without rating (you might need to modify your settings in _WooCommerce_ > _Settings_ > _Products_ > _General_ > _Product ratings_).
2. Create a post with a _Reviews by Product_ block.
3. Verify sorting by _Highest rating_ or _Lowest rating_ doesn't hide the reviews without rating.

Note: sorting by _Lowest rating_ shows reviews without rating first (see GIF above). I couldn't find any performant way to sort them after the other reviews, but it seems to me like an edge case. But suggestions are accepted. :slightly_smiling_face:

### CODE REVIEW

**Comment by Aljullu** [comment]
Thanks for the review @nerrad!

> What if you show the rating stars (but just empty) for reviews without ratings? It still retains the signal that it has no rating and doesn't make the sort appear buggy (depending on perspective).

If we do that, don't you think it might be confused with a review with 0 stars?

> I don't know if this is part of the spec or not, but it sounds like it be good to have an option for hiding reviews with no rating?

That's a use case we didn't account for. Do you mean allowing customers to hide reviews with no rating directly from the frontend? Or it should be a block attribute available in the editor?

**Comment by nerrad** [comment]
> If we do that, don't you think it might be confused with a review with 0 stars?

Hmm, I assumed that was functionally equivalent to "no rating" but if it's not that does affect things.  I could see how it might not be equivalent if someone doesn't want to leave a rating but does want to leave a review.

> Or it should be a block attribute available in the editor?

That :)

**Comment by Aljullu** [comment]
> Hmm, I assumed that was functionally equivalent to "no rating" but if it's not that does affect things. I could see how it might not be equivalent if someone doesn't want to leave a rating but does want to leave a review.

Well, there is the case that a store might have had ratings disabled (_WooCommerce_ > _Settings_ > _Products_ > _General_ > _Product ratings_ > _Enable star rating on reviews_) so users wrote reviews with no rating but eventually the store decided to enable ratings. In that case, I think it would be wrong showing the reviews with 0 stars because the users didn't have the opportunity to add any rating.

> That :)

cc'ing @joshuatf and @pmcpinto.

**tl;dr** should we add a toggle in the editor to hide reviews with no rating?

**Comment by nerrad** [comment]
> tl;dr should we add a toggle in the editor to hide reviews with no rating?

I forgot that showing ratings is an option - so this probably means this toggle should only appear IF that option is enabled.
